base/expr.jl

```diff
@@ -184,6 +184,57 @@ macro noinline(ex)
     esc(isa(ex, Expr) ? pushmeta!(ex, :noinline) : ex)
 end
 
+"""
+    @pure
+
+Annotates a function to ease type inference by strictly specifying the
+output is completely determined by the input.
+
+Note: Misuse can lead to regressions.
```

Comment:
I don’t know what regression means. This should just say “Usage can easily lead to whole program corruption or crashes and should be avoided.”

Reply:
Did you mean to say "Usage" rather than "Misuse"? 😂